src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,213 @@
+                    {
+                        tokenBuffer = new StringBuilder();
+                        isToken = true;
+                    }
+                    tokenBuffer.Append(currentCharacter);
+
+                    // Treat as escape character for }}
+                    while (position < length && resourceString[position] == '}')
+                    {
+                        tokenBuffer.Append(resourceString[position]);
+                        position++;
+                    }
+
+                    if (tokenBuffer != null && tokenBuffer.Length > 0)
+                    {
+                        var formattedString = string.Format(tokenBuffer.ToString(), arguments);
+                        if (formattedString.StartsWith("{") || formattedString.StartsWith("}"))
+                        {
+                            outputBuffer.Append(formattedString);
+                        }
+                        else
+                        {
+                            outputBuffer.Append(EncodeArgument(formattedString));
+                        }
+                    }
+
+                    if (position == length)
+                    {
+                        break;
+                    }
+                    else
+                    {
+                        currentCharacter = resourceString[position];
+                        position++;
+                        isToken = false;
+                        tokenBuffer = null;
+                    }
+                }
+
+                if (currentCharacter == '{')
+                {
+                    tokenBuffer = new StringBuilder();
+                    tokenBuffer.Append(currentCharacter);
+                    isToken = true;
+
+                    // Treat as escape character for {{
+                    while (position < length && resourceString[position] == '{')
+                    {
+                        tokenBuffer.Append(resourceString[position]);
+                        position++;
+                    }
+
+                    if (resourceString[position] < '0' || resourceString[position] > '9')
+                    {
+                        if (tokenBuffer != null && tokenBuffer.Length > 0)
+                        {
+                            outputBuffer.Append(string.Format(tokenBuffer.ToString()));
+                        }
+
+                        if (position == length)
+                        {
+                            break;
+                        }
+                        else
+                        {
+                            currentCharacter = resourceString[position];
+                            position++;
+                            isToken = false;
+                            tokenBuffer = null;
+                        }
+                    }
+                    else
+                    {
+                        //Remove the last added { since it is getting added in saveData condition below.
```

Comment:
What "saveData condition"? More generally, why does this block unconditionally add the extra "{"?